rules/consistent-destructuring.js

```diff
@@ -0,0 +1,140 @@
+const MESSAGE_ID_SUGGEST = 'consistentDestructuringSuggest';
+
+const declaratorSelector = [
+	'VariableDeclarator',
+	'[id.type="ObjectPattern"]',
+	'[init.type!="Literal"]'
+].join('');
+
+const isChildInParentScope = (child, parent) => {
+	while (child) {
+		if (child === parent) {
+			return true;
+		}
+
+		child = child.upper;
+	}
+
+	return false;
+};
+
+const fixDestructuring = (fixer, objectPattern, member, newMember) => {
+	// Check if member needs to be renamed
+	const property = member === newMember ?
+		`${member}` :
+		`${member}: ${newMember}`;
+	const {properties} = objectPattern;
+	const lastProperty = properties[properties.length - 1];
+
+	if (lastProperty) {
+		return fixer.insertTextAfter(lastProperty, `, ${property}`);
+	}
+
+	return fixer.replaceText(objectPattern, `{${property}}`);
+};
+
+const create = context => {
+	const {ecmaVersion} = context.parserOptions;
+	const source = context.getSourceCode();
+	const declarations = new Map();
+
+	return {
+		[declaratorSelector]: node => {
+			declarations.set(source.getText(node.init), {
+				scope: context.getScope(),
+				variables: context.getDeclaredVariables(node),
+				objectPattern: node.id
+			});
+		},
+		MemberExpression: node => {
+			const {parent, object, property} = node;
+
+			// Ignore member function calls and delete expressions
+			if (
+				(parent.type === 'CallExpression' &&
+				parent.callee === node) ||
+				(parent.type === 'UnaryExpression' &&
+				parent.operator === 'delete')
+			) {
+				return;
+			}
+
+			const declaration = declarations.get(source.getText(object));
+
+			if (!declaration) {
+				return;
+			}
+
+			const {scope, objectPattern} = declaration;
+			const memberScope = context.getScope();
+
+			// Property is destructured outside the current scope
+			if (!isChildInParentScope(memberScope, scope)) {
+				return;
+			}
+
+			const isNested = parent.type === 'MemberExpression';
+
+			if (isNested) {
+				context.report({
+					node,
+					messageId: MESSAGE_ID
+				});
+
+				return;
+			}
+
+			const destructurings = objectPattern.properties.filter(property =>
+				property.key.type === 'Identifier' && property.value.type === 'Identifier'
+			);
+			const expression = source.getText(node);
+			const member = source.getText(property);
+
+			// Member might already be destructured
+			const destructuredMember = destructurings.find(property =>
+				property.key.name === member
+			);
+			const newMember = destructuredMember ?
+				destructuredMember.value.name :
+				avoidCapture(member, [memberScope], ecmaVersion);
+
+			context.report({
+				node,
+				messageId: MESSAGE_ID,
+				data: {
+					expression,
+					property: newMember
+				},
+				suggest: [{
+					messageId: MESSAGE_ID_SUGGEST,
+					fix: destructuredMember ?
+						fixer => [fixer.replaceText(node, newMember)] :
+						fixer => [
+							fixer.replaceText(node, newMember),
+							fixDestructuring(fixer, objectPattern, member, newMember)
```

Comment:
Are you sure this is right? If multiple `MemberExpression`, will it update the `VariableDeclarator` again?

Reply:
It only fixes simple cases where the parent is not another `MemberExpression` (doesn't fix if `node.parent.type === 'MemberExpression'`), so I think it works correctly.

Reply:
I mean 

```js
const {a} =foo;
alert(foo.b);
alert(foo.b);
```

Reply:
It will give you two separate suggestions that will both add `b` to the destructured members and rename the member expression.